### PR TITLE
set xfail_strict=True

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
There is an `xfail_strict` pytest option, which makes xpass (passing tests that are marked as xfail) tests count as failures.

**Description of the Change:**
- Add `xfail_strict=true` to pytest.ini
- Fix all tests that were xpassing. I tried to make minimal changes to the tests themselves, but I'm open to more extensive test re-working if we think it's necessary

**Benefits:**
It's nice to have because it tells us if we've fixed old things we didn't know we've fixed, which should help us keep our testing etc. up to date

**Possible Drawbacks:**
If a test is decorated as `pytest.mark.xfail`, we seem to get the most out of this functionality. However, if you add a `pytest.xfail` WITHIN a test that would have passed afterwards, pytest doesn't know the difference and calls this a true xfail (when it should be a failing xpass)

[sc-52051]

### CODE REVIEW

File: tests/gradients/parameter_shift/test_parameter_shift.py [hunk 1]
```diff
@@ -3405,12 +3404,12 @@ def test_torch(self, dev_name, tol, broadcast):
...
-                NotImplementedError, match="Broadcasting with multiple measurements"
-            ):
-                res = self.cost_fn(weights, coeffs1, coeffs2, dev, broadcast)
-            return
+        # if broadcast:  # breaks xfail_strict
+        #     with pytest.raises(
+        #         NotImplementedError, match="Broadcasting with multiple measurements"
+        #     ):
+        #         res = self.cost_fn(weights, coeffs1, coeffs2, dev, broadcast)
+        #     return
```
Comment (albi3ro) [inline tests/gradients/parameter_shift/test_parameter_shift.py:3412]:
```
if broadcast:
    ....
    return 
pytest.xfail("some message")
```

only mark the non-broadcast case `xfail`.

Comment (timmysilv) [reply tests/gradients/parameter_shift/test_parameter_shift.py]: sure. it's kinda clumsy because the parametrization is for the whole class, and using `pytest.xfail` at some point within a test makes you miss the value-add of `xfail_strict`

File: tests/interfaces/default_qubit_2_integration/test_jax_jit_qnode_default_qubit_2.py [hunk 2]
```diff
@@ -1825,14 +1825,23 @@ def circuit():
 
         assert jax.numpy.allclose(circuit(), jax.numpy.array([1.0, 0.0]))
 
-    @pytest.mark.xfail(
-        reason="Non-trainable parameters are not being correctly unwrapped by the interface"
-    )
+    # @pytest.mark.xfail(
+    #     reason="Non-trainable parameters are not being correctly unwrapped by the interface"
+    # )
```
Comment (mudit2812) [inline tests/interfaces/default_qubit_2_integration/test_jax_jit_qnode_default_qubit_2.py:1830]:
```suggestion
```

File: tests/interfaces/test_jax_jit_qnode.py [hunk 3]
```diff
@@ -1385,7 +1385,7 @@ def circuit(x, y):
 
 
 # TODO: Add CV test when return types and custom diff are compatible
-@pytest.mark.xfail(reason="CV variables with new return types.")
+# @pytest.mark.xfail(reason="CV variables with new return types.")
```
Comment (mudit2812) [inline tests/interfaces/test_jax_jit_qnode.py:1388]:
```suggestion
```

File: tests/interfaces/test_jax_jit_qnode.py [hunk 4]
```diff
@@ -1870,14 +1870,20 @@ def circuit():
 
         assert jax.numpy.allclose(circuit(), jax.numpy.array([1.0, 0.0]))
 
-    @pytest.mark.xfail(
-        reason="Non-trainable parameters are not being correctly unwrapped by the interface"
-    )
+    # @pytest.mark.xfail(
+    #     reason="Non-trainable parameters are not being correctly unwrapped by the interface"
+    # )
```
Comment (mudit2812) [inline tests/interfaces/test_jax_jit_qnode.py:1875]:
```suggestion
```

**Review by mudit2812 — APPROVED** [review]
Agree about the clumsy changes to the parameter shift test, oh well.

File: tests/interfaces/default_qubit_2_integration/test_jax_jit_qnode_default_qubit_2.py (hunk 2 above)
Comment (timmysilv) [reply tests/interfaces/default_qubit_2_integration/test_jax_jit_qnode_default_qubit_2.py:1830]: wanted to keep this for context because i'm not entirely sure why certain cases still fail, and this comment might be helpful for someone in the future

File: tests/interfaces/test_jax_jit_qnode.py (hunk 4 above)
Comment (timmysilv) [reply tests/interfaces/test_jax_jit_qnode.py:1874]: same as other jax-jit comment

**Review by albi3ro — APPROVED** [review]
